torch/csrc/distributed/c10d/ProcessGroupXCCL.hpp

```diff
@@ -151,32 +163,60 @@ class TORCH_API ProcessGroupXCCL : public Backend {
       const std::vector<at::Tensor>& inputs = {},
       const std::vector<at::Tensor>& outputs = {});
 
-  template <typename Fn>
+  template <typename Fn, typename input_t, typename output_t>
+  c10::intrusive_ptr<Work> collective(
```

Comment:
Would you use the new `collective` definition in first PR? We don't want to change basic API in XCCL backend back and forth.

Reply:
rm allgatherv and align with nccl